### PR TITLE
Codefix for CS1591: Add ///<inheritdoc/>

### LINKED ISSUE #56620: Code fix for CS1591 to inherit documentation
A code fix should be offered for compiler warning CS1591 to apply an explicit `<inheritdoc/>` comment for members that implement or override another member. While this comment is generally unnecessary (https://github.com/dotnet/roslyn/issues/54927#issuecomment-918572803), build configurations that enable CS1591 may benefit from explicitly including this comment.

Users that disable CS1591 would not need or see this code fix.

### PR DESCRIPTION
**Pull request by MaStr11**
Fixes #56620

Tagging @sharwell

### CODE REVIEW

**Comment by MaStr11** [comment]
@sharwell Ready for review. If everything is settled, I will start to include VB support.

**Comment by MaStr11** [comment]
@sharwell and @CyrusNajmabadi This PR needs another review, because of major changes after the approval by @sharwell.

File: src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs
Comment (sharwell) [inline src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs:54]: ❔ Do we not have `GetRequiredSyntaxRootAsync` here? Current code is fine but will never return null for a C# document.

File: src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs
Comment (sharwell) [inline src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs:60]:
```suggestion
                if (!node.IsKind(SyntaxKind.MethodDeclaration, SyntaxKind.PropertyDeclaration, SyntaxKind.VariableDeclarator))
```

File: src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs
Comment (sharwell) [inline src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs:65]: This one can be slightly simplified

```suggestion
                if (node.IsKind(SyntaxKind.VariableDeclarator) && !node.Parent.IsParentKind(SyntaxKind.EventFieldDeclaration))
```

File: src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs
Comment (sharwell) [inline src/Features/CSharp/Portable/CodeFixes/AddInheritdoc/AddInheritdocCodeFixProvider.cs:107]:
```suggestion
                var inheritdocTagName = XmlName(DocumentationCommentXmlNames.InheritdocElementName).WithoutTrivia();
```

File: src/EditorFeatures/CSharpTest/Diagnostics/AddInheritdoc/AddInheritdocTests.cs
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddInheritdoc/AddInheritdocTests.cs:211]: 💡 Can you add tests showing the behavior for this for each of two cases:

1. The interface only declares `SomeEvent`
2. The interface declares both `SomeEvent` and `SomeOtherEvent`

```cs
public event System.Action SomeEvent, SomeOtherEvent;
```

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
@MaStr11 None of my comments are blocking, so I'm going to enable auto-merge. If you want to make changes based on my comments, you can do so here before the merge or submit a follow-up pull request (which would be small).

**Pull request merged**
